pkg/loader/executor.go

```diff
@@ -126,24 +125,40 @@ func (e *executor) updateMark(channel string, tx *tx) error {
 }
 
 // return a wrap of sql.Tx
-func (e *executor) begin() (*tx, error) {
+func (e *executor) begin() (*Tx, error) {
 	sqlTx, err := e.db.Begin()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	var tx = &tx{
+	var tx = &Tx{
 		Tx:                sqlTx,
 		queryHistogramVec: e.queryHistogramVec,
 	}
 
-	if e.info != nil && e.info.LoopbackControl {
-		err1 := e.updateMark("", tx)
-		if err1 != nil {
-			return nil, errors.Trace(err1)
+	if e.info.SupportPlugin {
+		isErr := false
+		hook := e.info.Hooks[plugin.LoaderPlugin]
+		hook.Range(func(k, val interface{}) bool {
+			c, ok := val.(LoopBack)
+			if !ok {
+				isErr = true
+				return false
+			}
+			tx = c.ExtendTxn(tx, e.info)
+			return true
+		})
+		if isErr {
+			return nil, errors.New("type is incorrect")
+		}
+	} else {
+		if e.info != nil && e.info.LoopbackControl {
```

Comment:
maybe `Support Plugin` should not conflict with `LoopbackControl`

Reply:
Thank you, I had fix it.